src/go/rpk/pkg/api/admin/api_config.go

```diff
@@ -65,3 +60,90 @@ func (a *AdminAPI) SetLogLevel(name, level string, expirySeconds int) error {
 	path := fmt.Sprintf("/v1/config/log_level/%s?level=%s&expires=%d", url.PathEscape(name), level, expirySeconds)
 	return a.sendOne(http.MethodPut, path, nil, nil)
 }
+
+type ConfigPropertyItems struct {
+	Type string `json:"type"`
+}
+
+type ConfigPropertyMetadata struct {
+	Type         string              `json:"type"`
+	Description  string              `json:"description"`
+	Nullable     bool                `json:"nullable"`
+	NeedsRestart bool                `json:"needs_restart"`
+	Visibility   string              `json:"visibility"`
+	Units        string              `json:"units,omitempty"`
+	Example      string              `json:"example,omitempty"`
+	Items        ConfigPropertyItems `json:"items,omitempty"`
+}
+
+type ConfigSchema map[string]ConfigPropertyMetadata
+
+type ConfigSchemaResponse struct {
+	Properties ConfigSchema `json:"properties"`
```

Comment:
Two reasons:
- to make the overall response object somewhat mirror the shape of a swagger 
- to future-proof in case we ever want to add a field to the response